### PR TITLE
fix vincenty for equatorial and coincident points

### LINKED ISSUE #418: Fail on vincenty distance
I am not expert in vincenty distance at all. And I was just testing how this functionality works, So I got a bug already on my second test. Code:

    let point1: Point<f64> = Point::new(0.0, 0.0);
    let point2: Point<f64> = Point::new(0.001, 0.0);
    let distance = point1.vincenty_distance(&point2).unwrap();

giving equal points will also give same error

    let point1: Point<f64> = Point::new(0.51, 88.1);
    let point2: Point<f64> = Point::new(0.51, 88.1);

The error:

    thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: FailedToConvergeError'

Looks like it panics when distance is 0?
I am not sure, but looks like it is wrong behavior.
Forward thank you!

    geo = "0.12.2"
    rustc 1.43.0-nightly

### PR DESCRIPTION
**Pull request by michaelkirk**
fixes #418 

From Algorithms for geodesics, Charles F. F. Karney∗
available https://arxiv.org/pdf/1109.4448.pdf

> For equatorial geodesics (φ1 = 0 and α1 = 1 2π), Eq. (11) is
> indeterminate; in this case, take σ1 = 0

Also fixes vincenty for coincident points.

### CODE REVIEW

File: geo/src/algorithm/vincenty_distance.rs [hunk 1]
```diff
@@ -196,4 +211,33 @@ mod test {
...
         );
     }
+
+    #[test]
+    fn test_vincenty_distance_equatorial() {
+        let a = Point::<f64>::new(0.0, 0.0);
+        let b = Point::<f64>::new(100.0, 0.0);
+        assert_relative_eq!(
+            a.vincenty_distance(&b).unwrap(),
+            11131949.079,
```
Comment (michaelkirk) [inline geo/src/algorithm/vincenty_distance.rs:221]: This value matches what comes out of geopy.vincenty 1.21.0

**Comment by urschrei** [comment]
lgtm! Could you leave a link to the PDF with the updated algorithm in the source, in case someone needs to go looking for it in the future?

**Comment by michaelkirk** [comment]
> lgtm! Could you leave a link to the PDF with the updated algorithm in the source, in case someone needs to go looking for it in the future?

Added! 

Please take another look @urschrei

**Comment by urschrei** [comment]
Thanks!

**Pull request merged**